homeassistant/core.py

```diff
@@ -1487,15 +1487,16 @@ async def async_call(
             _LOGGER.debug("Service was cancelled: %s", service_call)
             raise asyncio.CancelledError
         if task.done():
-            # Propagate any exceptions that might have happened during service call.
-            task.result()
+            # Get result and propagate any exceptions that might have happened during
+            # the service call.
+            result = task.result()
             # Service call completed successfully!
-            return True
+            return {"completed": True, "result": result}
         # Service call task did not complete before timeout expired.
         # Let it keep running in background.
         self._run_service_in_background(task, service_call)
         _LOGGER.debug("Service did not complete before timeout: %s", service_call)
-        return False
+        return {"completed": False}
```

Comment:
Suggestion:
```suggestion
        return {"completed": False, "result": none}
```
to keep the dict identical, which is easier handle in the upper layers.